src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,64 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_CAPACITY = 10;
+    private int size = 0;
+    private Cell[] cells = new Cell[STORAGE_CAPACITY];
+
     @Override
     public void put(K key, V value) {
+        int keyIndex;
+        if (size == 0 || (keyIndex = getIndexByKey(key)) < 0) {
```

Comment:
```suggestion
        int keyIndex = getIndexByKey(key);
        if (keyIndex < 0) {
```
if size == 0 getIndexByKey() would return -1